lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      false
+    end
```

Comment:
^See above for more.

So @byroot, how would you handle testing `SysVAtomicInteger` on a system that doesn't support semaphores? You would expect `SysVAtomicIntegerTest` to have assertions that things like
```
integer_1 = Semian::SysVAtomicInteger.new("example", 0660)
integer_2 = Semian::SysVAtomicInteger.new("example", 0660)
integer_2.value = 2
assert_equal(integer_2.value, integer_1.value)
```
That last line would definitely fail since you're running a fallback code while expecting it not to have fallback.